query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java

```diff
@@ -0,0 +1,65 @@
+package de.brandwatch.minianalytics.queryinput.controller;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import de.brandwatch.minianalytics.queryinput.postgres.model.Query;
+import de.brandwatch.minianalytics.queryinput.postgres.repository.QueryRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+
+@org.springframework.web.bind.annotation.RestController
+public class RestController {
+
+    Logger logger = LoggerFactory.getLogger(RestController.class);
+
+    @Autowired
+    QueryRepository queryRepository;
+
+    static final private Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
+    @RequestMapping(value = "/queries", method = RequestMethod.POST)
+    public ResponseEntity query(@RequestParam String query){
+        logger.info("API Call: query");
+        logger.info("Query: " + query);
+
+        //TODO parse Query into Lucene Query
```

Comment:
usually you would do that in a service layer. The usual architecture for apis like this is

Controller
Service
Repository

where the controller is only reponsible for handling the requests, possibly stuff like validation. The service is responsible for business logic, such as converting to a lucene query. And the repository is just responsible to provide methods to interact with any kind of storage.

This is also why Spring has the special annotations `@Controller`, `@Service`, `@Repository`. This automatically creates beans for those without you having to explicitly define them, and it helps to read code quicker because you always know where in the layer you are.

Of course not everything fits this pattern, which is why there is also the generic `@Component` I have mentioned before, or just defining the `@Bean` directly.

Long story short, this and the whole try catch block below propably belong into a `QueryService` class.

Reply:
Done